ipykernel/kernelspec.py

```diff
@@ -22,7 +22,7 @@
 RESOURCES = pjoin(os.path.dirname(__file__), 'resources')
 
 
-def make_ipkernel_cmd(mod='ipykernel', executable=None, extra_arguments=None, **kw):
+def make_ipkernel_cmd(mod='ipykernel_launcher', executable=None, extra_arguments=None, **kw):
```

Comment:
When you run `python -m ipykernel`, it loads `ipykernel/__init__.py` before `ipykernel/__main__.py`. The former already imports a bunch of stuff. Getting in before that happens would involve checks whenever `ipykernel` is imported, which seems like a bad thing.

With a separate module which is only useful as a means to start the kernel, we can check and modify `sys.path` before anything else is imported.